presto-elasticsearch/src/test/java/com/facebook/presto/elasticsearch/TestElasticsearchIntegrationSmoke.java

```diff
@@ -0,0 +1,89 @@
+import com.facebook.presto.testing.MaterializedResult;
+import com.facebook.presto.testing.MaterializedRow;
+import com.facebook.presto.testing.QueryRunner;
+import com.facebook.presto.tests.AbstractTestIntegrationSmokeTest;
+import io.airlift.tpch.TpchTable;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+
+import static com.facebook.presto.elasticsearch.EmbeddedElasticsearch.createEmbeddedElasticsearch;
+import static com.facebook.presto.testing.assertions.Assert.assertEquals;
+
+public class TestElasticsearchIntegrationSmoke
+        extends AbstractTestIntegrationSmokeTest
+{
+    public static final String SCHEMA = "tpch";
+    private final EmbeddedElasticsearch embeddedElasticsearch;
+
+    private QueryRunner queryRunner;
+
+    public TestElasticsearchIntegrationSmoke()
+            throws Exception
+    {
+        this(createEmbeddedElasticsearch());
+    }
+
+    public TestElasticsearchIntegrationSmoke(EmbeddedElasticsearch embeddedElasticsearch)
+    {
+        super(() -> ElasticsearchQueryRunner.createElasticsearchQueryRunner(embeddedElasticsearch, TpchTable.getTables()));
+        this.embeddedElasticsearch = embeddedElasticsearch;
+    }
+
+    @BeforeClass
+    public void setUp()
+    {
+        queryRunner = getQueryRunner();
+    }
+
+    @Test
+    @Override
+    public void testDescribeTable()
+    {
+        MaterializedResult actualColumns = this.computeActual("DESC orders").toTestTypes();
+        MaterializedResult.Builder builder = MaterializedResult.resultBuilder(this.getQueryRunner().getDefaultSession(), VarcharType.VARCHAR, VarcharType.VARCHAR, VarcharType.VARCHAR, VarcharType.VARCHAR);
+        for (MaterializedRow row : actualColumns.getMaterializedRows()) {
+            builder.row(row.getField(0), row.getField(1), "", "");
+        }
+        MaterializedResult filteredActual = builder.build();
+        builder = MaterializedResult.resultBuilder(this.getQueryRunner().getDefaultSession(), VarcharType.VARCHAR, VarcharType.VARCHAR, VarcharType.VARCHAR, VarcharType.VARCHAR);
+        MaterializedResult expectedColumns = builder
+                .row("orderkey", "bigint", "", "")
+                .row("custkey", "bigint", "", "")
+                .row("orderstatus", "varchar", "", "")
+                .row("totalprice", "double", "", "")
+                .row("orderdate", "varchar", "", "")
+                .row("orderpriority", "varchar", "", "")
+                .row("clerk", "varchar", "", "")
+                .row("shippriority", "bigint", "", "")
+                .row("comment", "varchar", "", "").build();
+        assertEquals(filteredActual, expectedColumns, String.format("%s != %s", filteredActual, expectedColumns));
+    }
+
+    @AfterClass(alwaysRun = true)
+    public final void destroy()
+            throws IOException
+    {
+        queryRunner.close();
```

Comment:
Use `Closer` here to make sure both resources had a chance to be closed.

Reply:
get it, thank you